velox/functions/prestosql/ArrayDistinct.cpp

```diff
@@ -140,6 +140,94 @@ class ArrayDistinctFunction : public exec::VectorFunction {
   }
 };
 
+template <>
+class ArrayDistinctFunction<UnknownType> : public exec::VectorFunction {
+ public:
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    auto& arg = args[0];
+
+    VectorPtr localResult;
+
+    // Input can be constant or flat.
+    if (arg->isConstantEncoding()) {
+      auto* constantArray = arg->as<ConstantVector<ComplexType>>();
+      const auto& flatArray = constantArray->valueVector();
+      const auto flatIndex = constantArray->index();
+
+      exec::LocalSingleRow singleRow(context, flatIndex);
+      localResult = applyFlat(*singleRow, flatArray, context);
+      localResult =
+          BaseVector::wrapInConstant(rows.end(), flatIndex, localResult);
+    } else {
+      localResult = applyFlat(rows, arg, context);
+    }
+
+    context.moveOrCopyResult(localResult, rows, result);
+  }
+
+ private:
+  VectorPtr applyFlat(
+      const SelectivityVector& rows,
+      const VectorPtr& arg,
+      exec::EvalCtx& context) const {
+    auto arrayVector = arg->as<ArrayVector>();
+    auto elementsVector = arrayVector->elements();
+    auto elementsRows =
+        toElementRows(elementsVector->size(), rows, arrayVector);
+    exec::LocalDecodedVector elements(context, *elementsVector, elementsRows);
+
+    vector_size_t elementsCount = elementsRows.end();
+    vector_size_t rowCount = rows.end();
+
+    // Allocate new vectors for indices, length and offsets.
+    memory::MemoryPool* pool = context.pool();
+    BufferPtr newIndices = allocateIndices(elementsCount, pool);
```

Comment:
The size of newIndices needed is at most rows.end() since every row at most has one element. So elementsCount and elementRows can be removed.